### PR TITLE
Select: updates story from knobs to control

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR updates story from knobs to control.
**Which issue(s) this PR fixes**:

Relates to #30182

### CODE REVIEW

File: packages/grafana-ui/src/components/Select/Select.story.tsx [hunk 1]
```diff
@@ -240,12 +247,13 @@ export const BasicSelectAsync = () => {
...
       }}
-      {...getDynamicProps()}
+      prefix={getPrefix(args.icon)}
+      {...args}
     />
   );
 };
 
-export const AutoMenuPlacement = () => {
+export const AutoMenuPlacement: Story = (args) => {
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Select/Select.story.tsx:256]: Not sure about this, but shouldn't this story have the menuplacement set to auto?

File: packages/grafana-ui/src/components/Select/Select.story.tsx [hunk 2]
```diff
@@ -276,14 +285,18 @@ export const CustomValueCreation = () => {
...
         }}
-        {...getDynamicProps()}
+        prefix={getPrefix(args.icon)}
+        {...args}
       />
     </>
   );
 };
+CustomValueCreation.args = {
+  allowCustomValue: false,
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Select/Select.story.tsx:301]: I think the whole point of this story is to show the custom creation option so maybe this should be
```suggestion
  allowCustomValue: true,
```

File: packages/grafana-ui/src/components/Select/Select.story.tsx [hunk 3]
```diff
@@ -17,10 +18,52 @@ export default {
...
+        'onCloseMenu',
+        'onCreateOption',
+        'onInputChange',
+        'onKeyDown',
+        'onOpenMenu',
+        'prefix',
+        'renderControl',
+        'options',
+        'isOptionDisabled',
+        'maxVisibleValues',
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Select/Select.story.tsx:46]: I think we can add `aria-label`, 'value' to this list too. Maybe the `noOptionsMessage` can be added as well as we don't have a story with no items to use it in, WDYT?

**Review by hugohaggmark — APPROVED** [review]
Looks good 🎉  left some nits for consideration.

File: packages/grafana-ui/src/components/Select/Select.story.tsx (hunk 3 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/Select/Select.story.tsx:46]: yeah...I agree. Will do that right away

File: packages/grafana-ui/src/components/Select/Select.story.tsx (hunk 2 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/Select/Select.story.tsx:301]: okidoki 👍

**Pull request merged**